src/mito2/src/region/options.rs

```diff
@@ -52,11 +55,13 @@ impl TryFrom<&HashMap<String, String>> for RegionOptions {
         let options: RegionOptionsWithoutEnum =
             serde_json::from_str(&json).context(JsonOptionsSnafu)?;
         let compaction: CompactionOptions = serde_json::from_str(&json).unwrap_or_default();
+        let wal_options: WalOptions = serde_json::from_str(&json).unwrap_or_default();
```

Comment:
I wonder if the serialization is compatible with old versions, or it's a breaking change.

Reply:
@killme2008 The deserialization scheme for `wal_options` is changed and it's compatible with the legacy RegionOptions. No need to add a compatible test since the existing tests already checked the compatibility.

Reply:
@v0y4g3r  The `try_from` implemented is changed. See: 
```rust
    fn try_from(options_map: &HashMap<String, String>) -> Result<Self> {
        let value = options_map_to_value(options_map);
        let json = serde_json::to_string(&value).context(JsonOptionsSnafu)?;

        // #[serde(flatten)] doesn't work with #[serde(default)] so we need to parse
        // each field manually instead of using #[serde(flatten)] for `compaction`.
        // See https://github.com/serde-rs/serde/issues/1626
        let options: RegionOptionsWithoutEnum =
            serde_json::from_str(&json).context(JsonOptionsSnafu)?;
        let compaction: CompactionOptions = serde_json::from_str(&json).unwrap_or_default();

        // Tries to decode the wal options from the map or sets to the default if there's none wal options in the map.
        let wal_options = options_map.get(WAL_OPTIONS_KEY).map_or_else(
            || Ok(WalOptions::default()),
            |encoded_wal_options| {
                serde_json::from_str(&encoded_wal_options).context(JsonOptionsSnafu)
            },
        )?;

        Ok(RegionOptions {
            ttl: options.ttl,
            compaction,
            storage: options.storage,
            wal_options,
        })
    }
```